@AH-64D Apache Longbow/Addons/fza_ah64_controls/scripting/functions/ihadss/fn_ihadssDraw.sqf

```diff
@@ -514,65 +514,39 @@ if (_was == WAS_WEAPON_MSL) then {
     };
     _targpos = _scPos;
 
-    if (isNull _mistargPos) then {
+    if (isNull _missileTarget || (_heli getVariable "fza_ah64_selectedMissile" == "fza_agm114l_wep" && isVehicleRadarOn _heli == false)) then {
         ((uiNameSpace getVariable "fza_ah64_raddisp") displayCtrl 131) ctrlSetText "";
     } else {
-        _terrainobscure = terrainIntersectasl[[(getPosASL _heli select 0) + ((sin getdir _heli) * 6), (getPosASL _heli select 1) + ((cos getdir _heli) * 6), (getPosASL _heli select 2)], [(getPosASL _mistargPos select 0), (getPosASL _mistargPos select 1), (getPosASL _mistargPos select 2) + 1]];
-        _obscureobjs = lineIntersectsWith[[(getPosASL _heli select 0) + ((sin getdir _heli) * 6), (getPosASL _heli select 1) + ((cos getdir _heli) * 6), (getPosASL _heli select 2)], getPosASL _mistargPos, _heli, _mistargPos];
-        _distOffAxis = abs ([[_heli, getPos _heli # 0, getPos _heli # 1, getPos _mistargPos # 0, getPos _mistargPos # 1] call fza_fnc_relativeDirection] call CBA_fnc_simplifyAngle180);
-        _targetDist = _heli distance getPos _mistargPos;
-	    _targ = _heli getVariable "fza_ah64_fcrNts";
+        _targetDist = _heli distance getPos _missileTarget;
+        _LoblCheckLima = [_heli, [getpos _missileTarget, "", speed _missileTarget, _missileTarget]] call fza_fnc_hellfireLimaLoblCheck;
+        _LoblCheck = [_heli] call fza_fnc_hellfireSalPrelaunch;
```

Comment:
Why are we always computing the checks for both? Can we do these inside the stages later for efficiency purposes?